editor/editor_properties_array_dict.cpp

```diff
@@ -230,8 +230,7 @@ void EditorPropertyArray::_change_type_menu(int p_index) {
 	Variant array = object->get_array().duplicate();
 	array.set(changing_type_index, value);
 
-	emit_changed(get_edited_property(), array, "", true);
-	update_property();
+	emit_changed(get_edited_property(), array);
```

Comment:
This changes a relevant part of the original code, was the removal of the arguments intentional? It's not the same as the rest, unsure if the `true` for `p_changing` is relevant here

Reply:
This is the main fix the argument here was preventing the updates from being from fired from the root Inspector which was the main reasons of the bug here. The rest is mainly preventing multiple updates and removing default arguments from function call ("" and false are the default values). I just realized with your comment that there was only here that it was to true and that therefore only changing type was not working.

Reply:
Do you mean the removal of `update_property`? Or did you just also remove the default arguments at the same time?

Reply:
the removal of update property is to prevent multiple updates. 
Removing the default argument ("" and false) does nothing and is just a by-product of how I made my search and replace.